wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -1194,11 +1198,118 @@ class PanelNode(BlockInclusionNode):
+    - `classname` - For legacy patterns requiring field-specific classes. Avoid if possible.
+    - `show_label` - Hide the label if it is rendered outside of the field.
+    - `id_for_label` - Manually set this this if the field’s HTML isn’t rendered by Django (for example hard-coded in HTML).
+        We add an id to the label so we can use it as a descriptor for the "Add comment" button.
+    - `sr_only_label` - Make the label invisible for all but screen reader users. Use this if the field is displayed without a label.
+    - `icon` - Some fields have an icon, though this is generally a legacy pattern.
+    - `help_text` - Manually set this if the field’s HTML is hard-coded.
+    - `help_text_id` - The help text’s id, necessary so it can be attached to the field with `aria-describedby`.
+    - `show_add_comment_button` - Display a comment control within Wagtail forms.
+    - `label_text` - Manually set this if the field’s HTML is hard-coded.
+    - `error_message_id` - ID of the error message container element.
+    """
+
+    label_for = id_for_label or (field and field.id_for_label) or ""
+
+    context = {
+        "classname": classname,
+        "show_label": show_label,
+        "sr_only_label": sr_only_label,
+        "icon": icon,
+        "show_add_comment_button": show_add_comment_button,
+        "error_message_id": error_message_id,
+        "label_for": label_for,
+        "label_id": f"{label_for}-label" if label_for else "",
+        "label_text": label_text or (field and field.label) or "",
+        "required": field and field.field.required,
+        "contentpath": field.name if field else "",
+        "help_text": help_text or (field and field.help_text) or "",
+    }
+
+    if help_text_id:
+        context["help_text_id"] = help_text_id
+    elif field and field.help_text and field.id_for_label:
+        context["help_text_id"] = f"{field.id_for_label}-helptext"
+    else:
+        context["help_text_id"] = ""
+
+    if field:
+        context["rendered_field"] = rendered_field or render_with_errors(field)
+        context[
+            "field_classname"
+        ] = f"w-field--{ fieldtype(field) } w-field--{ widgettype(field) }"
+
+        errors = field.errors
+        has_errors = bool(errors)
+        if has_errors and hasattr(field.field.widget, "render_with_errors"):
+            # field handles its own error rendering, so don't output them here
+            # (but still keep has_errors=True to keep the error styling)
+            errors = []
+
+        context["has_errors"] = has_errors
+        context["errors"] = errors
+    else:
+        context["rendered_field"] = rendered_field
+        context["field_classname"] = ""
+        context["has_errors"] = False
+        context["errors"] = []
+
+    return context
+
+
+@register.inclusion_tag("wagtailadmin/shared/formatted_field.html", takes_context=True)
+def formattedfieldfromcontext(context):
```

Comment:
The name gets quite long here, would something like `contextformattedfield` work better (to align with `rawformattedfield`)?

And I just want to point out that we have tags/filters with single "word" names, and we also have ones that use underscores. I don't really want to discuss which we should prefer right now though!